tests/test-single-user-mode.sh

```
@@ -0,0 +1,269 @@
+    local port="$1"
+    local extra_header="${2:-}"
+    local header_args=""
+    if [ -n "$extra_header" ]; then
+        header_args="-H \"$extra_header\""
+    fi
+
+    # Step 1: Initialize MCP session
+    eval curl -sf "http://localhost:$port/mcp" \
+        -H "'Content-Type: application/json'" \
+        -H "'Accept: application/json, text/event-stream'" \
+        $header_args \
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)